crates/electrum/tests/test_electrum.rs

```diff
@@ -559,3 +559,116 @@ fn test_sync_with_coinbase() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_check_fee_calculation() -> anyhow::Result<()> {
+    const SEND_AMOUNT: Amount = Amount::from_sat(10_000);
+    let env = TestEnv::new()?;
+    let electrum_client = electrum_client::Client::new(env.electrsd.electrum_url.as_str())?;
+    let client = BdkElectrumClient::new(electrum_client);
+
+    let spk_to_track = ScriptBuf::new_p2wsh(&WScriptHash::all_zeros());
+    let addr_to_track = Address::from_script(&spk_to_track, bdk_chain::bitcoin::Network::Regtest)?;
+
+    // Setup receiver.
+    let (mut recv_chain, _) = LocalChain::from_genesis_hash(env.bitcoind.client.get_block_hash(0)?);
+    let mut recv_graph = IndexedTxGraph::<ConfirmationBlockTime, _>::new({
+        let mut recv_index = SpkTxOutIndex::default();
+        recv_index.insert_spk((), spk_to_track.clone());
+        recv_index
+    });
+
+    // Mine some blocks.
+    env.mine_blocks(101, None)?;
+
+    // Send a preliminary tx such that the new utxo in Core's wallet
+    // becomes the input of the next tx
+    let new_addr = env
+        .rpc_client()
+        .get_new_address(None, None)?
+        .assume_checked();
+    let prev_amt = SEND_AMOUNT + Amount::from_sat(1650);
+    env.send(&new_addr, prev_amt)?;
+    let prev_hash_block = env.mine_blocks(1, None)?.into_iter().next();
+
+    let txid = env.send(&addr_to_track, SEND_AMOUNT)?;
+
+    // Mine a block to confirm sent tx.
+    let hash_block = env.mine_blocks(1, None)?.into_iter().next();
```

Comment:
nit: I would use `block_hash`, and `prev_block_hash` instead, but it's not a big deal. :)